### PR TITLE
Fixing doxygen warnings and ensuring valid doc and file names

### PR DESCRIPTION
**Pull request by rolanddenis**
# PR Description

:tada: This PR fixes all doxygen warnings :tada: 

Some remarks:
- the commit https://github.com/DGtal-team/DGtal/commit/7ecf025cb23c3c311d5a5236db57d07eed88a3f3 is the only one that modifies the compiled source. Doxygen seems to ignore the function overload in `testConvexHull2DReverse.cpp` because the return value is not assigned. I had to add dummy variables in order to fix these warnings.
- many warnings come from not documented parameters. I have try to fix this with appropriate comments but it was not always possible (see commit https://github.com/DGtal-team/DGtal/commit/17a302a5a897ad183e9fbc0efc33a0a1211d58b8 )
- some warnings comes from special characters that are not escaped, like < > (eg template) that is sometimes seen as an html tag (like \<I\>). They must be escaped with \
- all DGtal dependencies are now activated during documentation generation.
- macros coming from external libraries are not always correctly understood by doxygen. They must be added to the `PREDEFINED` doxygen parameter (see https://github.com/DGtal-team/DGtal/pull/1182/commits/34a27e35905556bbb9c4c7bb62bb6a411d52ebec ).

The next steps are to generate a Travis error if doxygen raises warnings and to ensure appropriate file name after \file or \example tag.

# Checklist

- [x] ~~Unit-test of your feature with [Catch](http://dgtal.org/doc/stable/moduleCatch.html).~~
- [x] Doxygen documentation of the code completed (classes, methods, types, members...)
- [x] Documentation module page added or updated.
- [x] New entry in the [ChangeLog.md](https://github.com/DGtal-team/DGtal/blob/master/ChangeLog.md) added.
- [x] No warning raised in Debug ```cmake``` mode (otherwise, Travis C.I. will fail).
- [x] All continuous integration tests pass (Travis & appveyor)
- [x] Travis checks that no doxygen warnings are raised :smiling_imp:

### CODE REVIEW

File: doc/doxy.config.in [hunk 1]
```diff
@@ -763,7 +763,7 @@ WARN_FORMAT            = "$file:$line: $text"
 # messages should be written. If left blank the output is written to standard
 # error (stderr).
 
-WARN_LOGFILE           =
+WARN_LOGFILE           = ./doxygen.log
```
Comment (dcoeurjo) [inline doc/doxy.config.in]: ``` @top_bindir@/doxygen.log``` ?

Comment (rolanddenis) [inline doc/doxy.config.in]: Right ! I'll also changed it for the other doxygen config files.

**Comment by dcoeurjo** [comment]
I'm on this PR.;)

**Comment by dcoeurjo** [comment]
@rolanddenis , I'll PR onto your branch. Can you please have a look ?

**Comment by dcoeurjo** [comment]
(and I have a regexp issue with the checking scripts you sent me)

**Comment by rolanddenis** [comment]
Ok, I'm looking at it, thanks !

**Comment by rolanddenis** [comment]
Question: I suppose that travis will take into account his new configuration only when this PR will be merged in DGtal ?

**Comment by dcoeurjo** [comment]
When the PR is merged, the doc is deployed. The documentation build and test scripts should be run even if the PR is pending.

**Comment by dcoeurjo** [comment]
@rolanddenis could you please merge the master back to your branch ?

File: .travis.yml [hunk 2]
```diff
@@ -100,7 +100,7 @@ script:
   - if [ $NEEDEXAMPLESANDTESTS == "true" ]; then  cd ../tests &&  make -j 3 &&  cd .. ; fi
   - if [ $NEEDEXAMPLESANDTESTS == "true" ]; then  make test ARGS=--output-on-failure ; fi
   - if [ $CONFIG == "DGtalTools" ]; then pwd ; .travis/getAndCheckDGtalTools.sh ; fi
-  - if [ $CONFIG == "Documentation,Debug,Cairo,GMP" ]; then make doc;  ./checkDoxygenDocumentation.sh; fi
+  - if [ $CONFIG == "Documentation,Debug,Cairo,GMP" ]; then make doc;  ./travis/checkDoxygenDocumentation.sh; fi
```
Comment (dcoeurjo) [inline .travis.yml]: ./.travis/ here

**Comment by dcoeurjo** [comment]
👏

**Comment by rolanddenis** [comment]
Yes, I need two commits to change a path ... :sweat:

**Comment by rolanddenis** [comment]
:scream: 
Do you have all those warnings on your computer ?!!

Many problems:
- doxygen on Travis checks the ih files and reads it as C++ code (on my computer, he doesn't associate .ih to C++) and he founds many errors (parameters with different names, doxygen bugs, ..). I will maybe remove *.ih from doxygen configuration, since those files are not visible in the final documentation ?
- template parameters, like `<T>`, trigger warnings (wrong xml/html tag). It is sometimes an issue with my doxygen version too. I will escape the `<>`.
- Travis's doxygen does not recognize some symbols, like `&alpha;` and `&pi;`. Newer versions of doxygen are OK with this...

Is it possible too use a newer version of doxygen ?

**Comment by dcoeurjo** [comment]
wait.. for the DOC target on travis, I manually download the last release (line 79)

we must do the same for the Documentation,...;GMP target (line 103)

**Comment by dcoeurjo** [comment]
```$DOC == true``` means "Documentation,Debug,Cairo,GMP" and PR merged to the master.

but we need the make doc (and doxygen checks) for all PR or branches (line 103)

**Comment by rolanddenis** [comment]
Ok, I see !

**Comment by rolanddenis** [comment]
Why `Debug,Cairo,GMP` ? It seems that this target only make the documentation ?

**Comment by rolanddenis** [comment]
It works (even if I've made a little mistake in the script) !! There is an error because doxygen doesn't download the DGtalTools tag-file.

**Comment by rolanddenis** [comment]
I have also clean the target name for a simple "Documentation". If the "Cairo,GMP" was here to include these extensions in the documentation, I have already activated all extensions in the doxygen config file (see https://github.com/DGtal-team/DGtal/pull/1182/files#diff-8fde8a3a5ed4792d591a704551bf7fedR2048 ).

**Comment by rolanddenis** [comment]
:tada:

**Comment by dcoeurjo** [comment]
🎉🎉🎉

**Pull request merged**